### PR TITLE
Restore SlimPajama preprocessing code

### PR DESCRIPTION
**Pull request by aflah02**
Rework of https://github.com/Lightning-AI/litgpt/pull/1838

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -42,7 +42,7 @@ all = [
     "sentencepiece>=0.2.0",      # llama-based models
     "requests>=2.31.0",          # litgpt.data
     "litdata==0.2.17",           # litgpt.data
-    "litserve>=0.1.5",           # litgpt.deploy
+    "litserve>=0.2.4",           # litgpt.deploy
```
Comment (Andrei-Aksionov) [inline pyproject.toml:45]:
```suggestion
    "litserve<=0.2.4",           # litgpt.deploy
```

**Comment by aflah02** [comment]
Thanks @Andrei-Aksionov 
It seems this is ready

**Comment by Andrei-Aksionov** [comment]
Thanks for the patience @aflah02

**Comment by aflah02** [comment]
haha thanks for helping me debug :)

**Pull request merged**